doc/reactions.md

```diff
@@ -0,0 +1,152 @@
+
+Reactions are defined in JSON:
+
+``` json
+
+{
+  "identityPaths": [
+    [
+      "actor",
+      "mbox"
+    ]
+  ],
+  "conditions": {
+    "a": {
+      "and": [
+        {
+          "path": [
+            "object",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/activities/a"
+        },
+        {
+          "path": [
+            "verb",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/verbs/completed"
+        },
+        {
+          "path": [
+            "result",
+            "success"
+          ],
+          "op": "eq",
+          "val": true
+        }
+      ]
+    },
+    "b": {
+      "and": [
+        {
+          "path": [
+            "object",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/activities/b"
+        },
+        {
+          "path": [
+            "verb",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/verbs/completed"
+        },
+        {
+          "path": [
+            "result",
+            "success"
+          ],
+          "op": "eq",
+          "val": true
+        },
+        {
+          "path": [
+            "timestamp"
+          ],
+          "op": "gt",
+          "ref": {
+            "condition": "a",
+            "path": [
+              "timestamp"
+            ]
+          }
+        }
+      ]
+    }
+  },
+  "template": {
+    "actor": {
+      "mbox": {
+        "$templatePath": [
+          "a",
+          "actor",
+          "mbox"
+        ]
+      }
+    },
+    "verb": {
+      "id": "https://example.com/verbs/completed"
+    },
+    "object": {
+      "id": "https://example.com/activities/a-and-b",
+      "objectType": "Activity"
+    }
+  }
+}
+
+
+```
+
+### Identity Paths
+
+`identityPaths` is an array of zero or more paths that must match across all statements in the sequence. These could include an actor identifier or statement registration.
+
+### Conditions
+
+`conditions` is a mapping of names to rules for finding significant statements. Rules can be composed with boolean logic.
+
+In the example given above statement `a` must have an object id equal to `https://example.com/activities/a`, a verb id equal to `https://example.com/verbs/completed`, and a result success equal to `true`. Statement `b` must have the same verb and result success but an object id equal to `https://example.com/activities/b` and a timestamp greater than that of `a`.
+
+#### Rules
+
+All rules have a `path` array that indicates a path in an xAPI statement and an `op` that is one of the following operators:
+
+* `gt` - Greater than
+* `lt` - Less than
+* `gte` - Greater than or equal
+* `lte` - Less than or equal
+* `eq` - Equal
+* `noteq` - Not equal
+* `like` - Fuzzy match using SQL `%` syntax
```

Comment:
Might want to link to an example of `%` usage (I'm not familiar with it, having never used it before).

Reply:
this is valid. We'll have some non-SQL users.

Reply:
I just inlined it. https://github.com/yetanalytics/lrsql/pull/306/commits/2e77a64a568fdcb71eded7e34abad0d5ff07cc98